app/authorization.py

```diff
@@ -0,0 +1,80 @@
+"""Authorization"""
+
+import json
+import os
+import secrets
+from typing import Optional
+
+from constants import (
+    AUTH_ENABLED_KEY,
+    PASSWORD_KEY,
+    USER_KEY,
+    AuthTypes,
+    ExceptionMessages,
+)
+from fastapi import Depends, HTTPException, status
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+
+AUTH_ENABLED = os.getenv(AUTH_ENABLED_KEY, "false").lower() == "true"
+security = HTTPBasic() if AUTH_ENABLED else lambda: None
+
+
+def get_authorization(credentials: Optional[HTTPBasicCredentials] = Depends(security)):
+    """Helper function to validate user authorization"""
+    if credentials is None:
+        return AuthTypes.NO_AUTH.value
+
+    if not is_valid_request(credentials):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail=ExceptionMessages.UNAUTHORIZED.value,
+            headers={"WWW-Authenticate": "Basic"},
+        )
+
+    return AuthTypes.HTTP_BASIC.value
```

Comment:
This should be the type returned as well.

Reply:
return the current active auth mode